### PR TITLE
API Definition in OpenAPI (Swagger)

### PR DESCRIPTION
**Pull request by mvamgc**
should be merged to backend service

### CODE REVIEW

**Comment by mvamgc** [comment]
Use Swagger Editor for review https://github.com/swagger-api/swagger-editor

File: docs/api.yaml [hunk 1]
```diff
@@ -0,0 +1,270 @@
+openapi: 3.0.0
+info:
+  title: E-Commerce
+  description: 'E-Commerce API define functionality required for a small online store. It allows search products, add to a shopping card and checkout the products in the cart.'
```
Comment (IgorKulishov) [inline docs/api.yaml:4]: Is it ok that we currently do not have search funct-ty?

File: docs/api.yaml [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+  version: '1.0'
+  contact: {}
+tags:
+  - name: product
+    description: 'responsible for list of products, product categories and search'
+  - name: cart
+    description: |
+      Shopping cart functionality.
+      Content of shopping cart can be connected to account or to a session ID. If user is not logged it, he can still
```
Comment (IgorKulishov) [inline docs/api.yaml:13]: It's nice feature " If user is not logged it, he can still work with the site", but still we do not have it yet. 
Is it ok to add it to swagger?

File: docs/api.yaml (hunk 1 above)
Comment (mvamgc) [reply docs/api.yaml:4]: Search functionality is the most important for online shop. You can drop all other functionality, and leave search only and still run business.

File: docs/api.yaml (hunk 2 above)
Comment (mvamgc) [reply docs/api.yaml:13]: Shopping card is not documented at all. I do not have any information on it. What do you suggest ?

Comment (IgorKulishov) [reply docs/api.yaml:13]: We do have short description https://github.com/IgorKulishov/ecommerce-ui-template#checkout-products-in-shopping-cart

File: docs/api.yaml (hunk 1 above)
Comment (IgorKulishov) [reply docs/api.yaml:4]: I suggest to move all code to the API repository https://github.com/IgorKulishov/ecommerce-lambda-api

**Pull request closed without merging**